game-core/src/main/java/games/strategy/triplea/ResourceLoader.java

```diff
@@ -228,9 +228,29 @@ public boolean hasPath(final String path) {
    */
   public @Nullable URL getResource(final String inputPath) {
     final String path = resourceLocationTracker.getMapPrefix() + inputPath;
-    return getMatchingResources(path).stream()
-        .findFirst()
-        .orElse(getMatchingResources(inputPath).stream().findFirst().orElse(null));
+    return findResource(path).or(() -> findResource(inputPath)).orElse(null);
+  }
+
+  /**
+   * Returns the URL of the resource at the specified path or {@code null} if the resource does not
+   * exist. Tries the given 2 paths in order first in the map resources then engine resources.
+   *
+   * @param inputPath (The name of a resource is a '/'-separated path name that identifies the
+   *     resource. Do not use '\' or File.separator)
+   * @param inputPath2 Same as inputPath but this takes second priority when loading
+   */
+  public @Nullable URL getResource(final String inputPath, final String inputPath2) {
```

Comment:
I think the only way to take this further is have the single arg version call the new double arg or combine them together to take vararg or a list. That would increase the complexity of the new method since you need to handle a null or empty second parameter then. My general feel is there isn't much left to be gained here and the code is pretty clean after the helper method addition.